drivers/net/mlx5/mlx5_rxtx.c

```diff
@@ -1021,9 +1021,11 @@ mlx5_tx_burst_ipoib(void *dpdk_txq, struct rte_mbuf **pkts, uint16_t pkts_n)
 		length -= pkt_inline_sz;
 		addr += pkt_inline_sz;
 		/* Should we enable HW CKSUM offload. */
+		if (buf->ol_flags & PKT_TX_IP_CKSUM)
+			cs_flags = MLX5_ETH_WQE_L3_CSUM;
 		if (buf->ol_flags &
-			(PKT_TX_IP_CKSUM | PKT_TX_TCP_CKSUM | PKT_TX_UDP_CKSUM))
-			cs_flags = MLX5_ETH_WQE_L3_CSUM | MLX5_ETH_WQE_L4_CSUM;
+			(PKT_TX_TCP_CKSUM | PKT_TX_UDP_CKSUM))
```

Comment:
you can always do the L3 csum. there is no perf penalty in HW. 

in other words - can we keep the code as it was or there is a bug here?